File: 1-twoSum/Go/twoSum.go
package twoSum

func TwoSum(nums []int, target int) []int {
	var smap = make(map[int]int)
	for i, num := range nums {
		x := target - num
		if index, ok := smap[x]; ok {
			return []int{index, i}
		}
		smap[num] = i
	}
	return []int{}
}

File: 1-twoSum/python3/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen_dict = {}
        for i, num in enumerate(nums):
            x = target - num
            if x in seen_dict.keys():
                return [seen_dict[x], i]
            seen_dict[num] = i
        
        